reana_client/utils.py

```diff
@@ -173,6 +180,57 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def _validate_environment(reana_yaml):
+    """Validate environments in REANA specification file according to workflow type.
+
+    :param reana_yaml: Dictionary which represents REANA specifications file.
+    """
+    workflow_type = reana_yaml["workflow"]["type"]
+
+    if workflow_type == "serial":
+        workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
+        _validate_serial_workflow_environment(workflow_steps)
+    elif workflow_type == "yadage":
+        logging.warning(
+            "The environment validation is not implemented for Yadage workflows yet."
+        )
+    elif workflow_type == "cwl":
+        logging.warning(
+            "The environment validation is not implemented for CWL workflows yet."
+        )
+
+
+def _validate_serial_workflow_environment(workflow_steps):
+    """Validate environments in REANA serial workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if valid tags are not used in serial workflow steps.
```

Comment:
This function will perform more validations so we can just have: "Warns user if the workflow environment is invalid in serial workflow steps."